unity-renderer/Assets/DCLServices/MapRendererV2/MapRenderer.cs

```diff
@@ -167,6 +167,25 @@ private void DisableLayers(MapLayer mask)
             }
         }
 
+        public void ToggleLayer(MapLayer mask, bool isActive)
+        {
+            foreach (MapLayer mapLayer in ALL_LAYERS)
+            {
+                if (EnumUtils.HasFlag(mask, mapLayer)
+                    && layers.TryGetValue(mapLayer, out var mapLayerStatus))
+                {
+                    if (isActive)
+                    {
+                        mapLayerStatus.MapLayerController.Enable(mapLayerStatus.CTS.Token).SuppressCancellationThrow().Forget();
+                    }
+                    else
+                    {
+                        mapLayerStatus.MapLayerController.Disable(mapLayerStatus.CTS.Token).SuppressCancellationThrow().Forget();
+                    }
+                }
+            }
+        }
+
```

Comment:
this one is important ‼️ - you need to refactor it to use new sharedActivity property of MapLayerStatus (basically you can use here SetSharedActivity instead of direct Enable/Disable)